compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -1266,18 +1266,29 @@ pub trait PrettyPrinter<'tcx>:
             ty::Ref(
                 _,
                 Ty(Interned(
```

Comment:
Maybe use nested match with the inner match matching on `pointee.kind()`? This indentation is getting crazy and pretty unreadable.